mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/index/Restaurant.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.mongodb.test.index;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Index;
+import javax.persistence.Table;
+
+import org.hibernate.ogm.datastore.mongodb.type.GeoPoint;
+import org.hibernate.ogm.options.shared.IndexOption;
+import org.hibernate.ogm.options.shared.IndexOptions;
+
+@Entity
+@Table(name = Restaurant.COLLECTION_NAME, indexes = {
+		@Index(columnList = "location", name = "location_spatial_idx")
+})
+@IndexOptions(
+		@IndexOption(forIndex = "location_spatial_idx", options = "{ _type: '2dsphere' }")
+)
```

Comment:
As an evolution, would it be possible to deduce this from the fact that we are using a GeoPoint?

Reply:
Yes, it would be possible to have some sort of default type if the property is a GeoPoint.